ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Sandboxed/BraveTranslateScript.js

```diff
@@ -0,0 +1,8438 @@
+window.__firefox__ = {};
```

Comment:
I went ahead and found each script + dependency manually in Chromium. Then I got the un-obfuscated version of those scripts and replaces the necessary code with it + added links to where the script was found.

This should drastically help as there's no more obfuscated JS.

Reply:
The current approach looks extremely hard to support and review.
It's not clear why we have to bundle all the stuff into a one script. Why we can't use the desktop approach?
Also, the script from [translate.brave.com](https://github.com/brave/go-translate/tree/master/assets/static/v1) is designed to be updatable from the backend side.

Option 1 (preferable): enable translate code from Chromium for iOS (or at least partially): https://source.chromium.org/chromium/chromium/src/+/main:components/translate/ios/browser/
It probably needs some chromium primitives like WebState, not sure about the current status wherever we support them or not.

Option 2: bundle all the supplementary scripts from b-c using webpack and put them to resources. Load the main script from https://translate.brave.com/, combine with the new script from resources and inject

@kylehickinson @StephenHeaps  maybe you have ideas how to simplify things here?

Reply:
1. We don't support WebState yet, so for this PR it isn't possible to use the Desktop approach.
2. If we load it from translate.brave.com, it doesn't help that it's still minified and unreadable and still impossible to review.
3. Loading the scripts from translate.brave.com doesn't let us modify it to add Apple Translate support, which this PR does.
4. How do we web-pack all the Chromium translate scripts and dependencies including all the ones specific to iOS that are in typescript, with Webpack?

Reply:
thanks for the aswers, @Brandon-T.

>We don't support WebState yet, so for this PR it isn't possible to use the Desktop approach.
Ok, we can't reuse it as-is, but I mean to reuse the approach to build the script on the fly.

>If we load it from translate.brave.com, it doesn't help that it's still minified and unreadable and still impossible to review.
It makes sense for the translate script itself, but not for all the script from /src. 
The point is to have the same script for desktop & iOS, but different platform adapters.

If you want to improve the current translate script, it's stored here: https://github.com/brave/go-translate/tree/master/assets/static/v1

>Loading the scripts from translate.brave.com doesn't let us modify it to add Apple Translate support
it's not clear to me.. If we download the script in advance and store locally, what is the difference with hardcoded script?

> 4. How do we web-pack all the Chromium translate scripts and dependencies including all the ones specific to iOS that are in typescript, with Webpack?
Why we can't import them in a .ts file to transpile. That way we avoid duplicating the files.
Otherwise it's not clear how they will be synchronized in future.
Transpiling via `transpile_web_ui ` + loading via a resource bundle ([example](https://github.com/brave/brave-core/blob/master/ios/browser/api/brave_wallet/brave_wallet_api.mm#L125)) works for iOS.

Reply:
@atuchin-m 

1. I download it on the fly in the latest commit. So now it downloads the translate script instead of hard-coding.
2. `transpile_web_ui` only works with WebUI. It doesn't work with just plain .ts files without any html.

https://github.com/brave/reviews/issues/1650#issuecomment-2223144564

It requires us to build resource files for iOS to export strings and other resources explicitly, when WebUI is not involved. We've had this discussion with Brian J. and a few others. Also it currently breaks unit tests. It's on the roadmap, but it's not ready yet, so for now it's not possible.

Reply:
> Also it currently breaks unit tests.

Not sure specifics of what unit tests would break / how in this PR, but this sounds similar to unit tests in https://github.com/brave/brave-core/pull/25694. I will be updating that PR soon to allow loading the needed resources from BraveCore (we need to load `procedural_filters.ts` to allow our `ScriptExecutionTests` to use it & run again, some more info [here](https://bravesoftware.slack.com/archives/C07N27PH5PZ/p1727268068713789)). Not sure if that approach helps with unit test concerns here too.